Source/Scene/ModelExperimental/ModelExperimentalPrimitive.js

```diff
@@ -117,9 +120,18 @@ function initialize(runtimePrimitive) {
   var hasQuantization = ModelExperimentalUtility.hasQuantizedAttributes(
     primitive.attributes
   );
+  var is3DTiles = ModelExperimentalType.is3DTiles(model.type);
+  var hasPointCloudShading = is3DTiles || defined(model.pointCloudShading);
 
   pipelineStages.push(GeometryPipelineStage);
 
+  if (
+    hasPointCloudShading &&
+    primitive.primitiveType === PrimitiveType.POINTS
+  ) {
+    pipelineStages.push(PointCloudAttenuationPipelineStage);
```

Comment:
This pipeline-configuration code should be split out into its own function so it can be reconfigured from `ModelExperimentalSceneGraph.buildDrawCommands()` when necessary.